x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/find_gaps.ts

```diff
@@ -59,3 +59,54 @@ export const findGaps = async ({
     throw err;
   }
 };
+
+/**
+ * This function is used to find gaps using search after.
+ * It's used when to be able process more than 10,000 gaps with stable sorting.
+ */
+export const findGapsSearchAfter = async ({
+  eventLogClient,
+  logger,
+  params,
+}: {
+  eventLogClient: IEventLogClient;
+  logger: Logger;
+  params: FindGapsSearchAfterParams;
+}): Promise<{
+  total: number;
+  data: Gap[];
+  searchAfter?: SortResults[];
+  pitId?: string;
+}> => {
+  const { ruleId, start, end, perPage, statuses, sortField, sortOrder } = params;
+  try {
+    const filter = buildGapsFilter({ start, end, statuses });
+    const gapsResponse = await eventLogClient.findEventsBySavedObjectIdsSearchAfter(
+      RULE_SAVED_OBJECT_TYPE,
+      [ruleId],
+      {
+        filter,
+        sort: [
+          {
+            sort_field: sortField ?? '@timestamp',
+            sort_order: sortOrder ?? 'desc',
+          },
+        ],
+        per_page: perPage,
+        pit_id: params?.pitId,
+        search_after: params?.searchAfter,
+      }
+    );
+
+    return {
+      total: gapsResponse.total,
+      data: transformToGap(gapsResponse),
+      searchAfter: gapsResponse.search_after,
+      pitId: gapsResponse.pit_id,
+    };
+  } catch (err) {
+    // Clean up PIT if there's an error and we created it
```

Comment:
should we call `eventLogClient.closePointInTime` as per the comment here?

Reply:
It's from initial implementation - I removed comment.

Now we are closing PIT in   `queryEventsBySavedObjectsSearchAfter` in case of a error